src/sentry/api/helpers/group_index/update.py

```diff
@@ -644,16 +644,10 @@ def update_groups(
     elif result.get("hasSeen") is False:
         GroupSeen.objects.filter(group__in=group_ids, user_id=acting_user.id).delete()
 
-    if result.get("isBookmarked"):
-        for group in group_list:
-            GroupBookmark.objects.get_or_create(
-                project=project_lookup[group.project_id], group=group, user_id=acting_user.id
-            )
-            GroupSubscription.objects.subscribe(
-                user=acting_user, group=group, reason=GroupSubscriptionReason.bookmark
-            )
-    elif result.get("isBookmarked") is False:
-        GroupBookmark.objects.filter(group__in=group_ids, user_id=acting_user.id).delete()
+    if "isBookmarked" in result:
+        handle_is_bookmarked(
+            result.get("isBookmarked"), group_list, group_ids, project_lookup, acting_user
```

Comment:
```suggestion
            result["isBookmarked"], group_list, group_ids, project_lookup, acting_user
```

Since you first do `isBookmarked" in result`.